src/phoenix/server/api/types/PromptVersionTemplate.py

```diff
@@ -0,0 +1,89 @@
+# Part of the Phoenix PromptHub feature set
+
+from enum import Enum
+from typing import Any, Union
+
+import strawberry
+from pydantic import BaseModel
+from strawberry.scalars import JSON
+
+JSONSerializable = Union[None, bool, int, float, str, dict[str, Any], list[Any]]
+
+
+@strawberry.enum
+class PromptMessageRole(str, Enum):
+    USER = "user"
+    SYSTEM = "system"
+    AI = "ai"  # E.g. the assistant. Normalize to AI for consistency.
+
+
+class PromptStringTemplate(BaseModel):
+    template: str
+
+
+class TextPromptMessage(BaseModel):
+    role: PromptMessageRole
+    content: str
+
+    def to_gql(self):
+        return TextPromptMessageGQL.from_model(self)
+
+
+class JSONPromptMessage(BaseModel):
```

Comment:
Do we need both to and from methods?

Reply:
maybe eventually but we're feeling it out